jolt-core/src/zkvm/prover.rs

```diff
@@ -1543,70 +1769,259 @@ mod tests {
-        populate_memory_states(0, &trusted_advice, Some(&mut trusted_advice_words), None);
-        let trusted_advice_commitment = {
-            crate::poly::commitment::dory::DoryGlobals::initialize_trusted_advice(
-                1,
-                (max_trusted_advice_size as usize) / 8,
-            );
-            let _ctx = crate::poly::commitment::dory::DoryGlobals::with_context(
-                crate::poly::commitment::dory::DoryContext::TrustedAdvice,
-            );
-            let poly = MultilinearPolynomial::<ark_bn254::Fr>::from(trusted_advice_words);
-            let (trusted_advice_commitment, _hint) =
-                <crate::poly::commitment::dory::DoryCommitmentScheme as CommitmentScheme>::commit(
-                    &poly,
-                    &preprocessing.generators,
-                );
-            trusted_advice_commitment
-        };
+        let (trusted_commitment, trusted_hint) =
+            commit_trusted_advice_preprocessing_only(&preprocessing, &trusted_advice);
 
         let prover = RV64IMACProver::gen_from_elf(
             &preprocessing,
-            elf_contents,
+            &elf_contents,
             &inputs,
             &untrusted_advice,
             &trusted_advice,
-            Some(trusted_advice_commitment),
+            Some(trusted_commitment),
+            Some(trusted_hint),
         );
         let io_device = prover.program_io.clone();
         let (jolt_proof, debug_info) = prover.prove();
 
         let verifier_preprocessing = JoltVerifierPreprocessing::from(&preprocessing);
-        let verifier = RV64IMACVerifier::new(
+        RV64IMACVerifier::new(
             &verifier_preprocessing,
             jolt_proof,
             io_device.clone(),
-            Some(trusted_advice_commitment),
+            Some(trusted_commitment),
             debug_info,
         )
-        .unwrap();
-        let verification_result = verifier.verify();
-        assert!(
-            verification_result.is_ok(),
-            "Verification failed with error: {:?}",
-            verification_result.err()
+        .expect("Failed to create verifier")
+        .verify()
+        .expect("Failed to verify proof");
+
+        // Verify output is correct (advice should not affect sha2 output)
+        let expected_output = &[
+            0x28, 0x9b, 0xdf, 0x82, 0x9b, 0x4a, 0x30, 0x26, 0x7, 0x9a, 0x3e, 0xa0, 0x89, 0x73,
+            0xb1, 0x97, 0x2d, 0x12, 0x4e, 0x7e, 0xaf, 0x22, 0x33, 0xc6, 0x3, 0x14, 0x3d, 0xc6,
+            0x3b, 0x50, 0xd2, 0x57,
+        ];
+        assert_eq!(io_device.outputs, expected_output);
+    }
+
+    #[test]
+    #[serial]
+    fn max_advice_with_small_trace() {
+        // Tests that max-sized advice (4KB = 512 words) works with a minimal trace.
+        // With balanced dims (sigma_a=5, nu_a=4 for 512 words), the minimum padded trace
+        // (256 cycles -> total_vars=12) is sufficient to embed advice.
```

Comment:
what happens when the advice exceeds the max size? Maybe we should have a `#[should_panic]` test for that edge case?

Reply:
The max-sized here refers to the `max_trusted_advice_size` of memory layout config. There is no actual maximum length. I added two tests for it anyways.